File: python/execution-timeout/handler.py
from flask import jsonify
import time

def handler(request):
    try:
        print("Starting execution...")
        time.sleep(1)  # Delay for 1 second
        print("Execution resumed after delay.")
        return jsonify({'message': 'Function executed with intentional delay.'})
    except Exception as e:
        print(f"Error: {e}")
        return jsonify({
            "statusCode": 500,
            "headers": {"Content-Type": "text/plain"},
            "message": "Internal Server Error"
        })

File: nodejs/execution-timeout/handler.js
export const handler = async (req, res) => {
  try {
    // Set a timeout delay in milliseconds
    const delay = 1000; // 1 second

    // Create a promise that resolves after the delay
    const timeoutPromise = () =>
      new Promise((resolve) => setTimeout(resolve, delay));

    console.log("Starting execution...");

    // Wait for the delay to pass
    await timeoutPromise();

    console.log("Execution resumed after delay.");

    const responseJson = {
      message: "Function executed with intentional delay.",
    };

    // Set the response headers
    res.setHeader("Content-Type", "application/json");

    // Send the response JSON
    res.end(JSON.stringify(responseJson));
  } catch (error) {
    // Handle errors
    console.error(error);
    // Send an error response if needed
    res.statusCode = 500;
    res.setHeader("Content-Type", "text/plain");
    res.end("Internal Server Error");
  }
};
